### PR TITLE
Add Personal Information section with name, phone number, and email address

### LINKED ISSUE #1: Ability to add users information to the top of the resume
https://github.com/MLH-Fellowship/orientation-project-python/issues/22

### PR DESCRIPTION
**Pull request by Epic-Eric**
##Work in progress
Fixes #1 
This adds a "Personal Information" section at the beginning of the page with name, phone number, and email address.
However, due to limited knowledge on React, I was not able to make far in the project.
I will try my best to contribute and improve the quality of my code.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -110,6 +110,17 @@ function App() {
...
+        <h2>Personal Information</h2>
+        <label for="name">Enter your full name: </label>
+        <input type="text" class="name"></input>
+        <br></br>
+        <label for="email">Enter your phone number: </label>
+        <input type="email" class="email"></input>
+        <br></br>
+        <label for="email">Enter your email address: </label>
+        <input type="tel" class="phoneNumber"></input>
+      </div>
```
Comment (Ahmedsaed) [inline src/App.js:123]:
```suggestion
      <from className="resumeSection">
        <h2>Personal Information</h2>
        <label for="name">Enter your full name: </label>
        <input type="text" class="name"></input>
        <br></br>
        <label for="email">Enter your phone number: </label>
        <input type="email" class="email"></input>
        <br></br>
        <label for="email">Enter your email address: </label>
        <input type="tel" class="phoneNumber"></input>
        <button type="submit">Save</button>
      </from>
```

# Changes
1. wrap the inputs in a form to handle validation and submission
2. add submit button

# TODO
1. Style the inputs
2. write the js code to handle state and submission

> [!NOTE]
> This is a suggestion. I believe you can accept it directly from GitHub and the modifications will be applied to the code.

**Review by Ahmedsaed — CHANGES_REQUESTED** [review]

**Comment by Ahmedsaed** [comment]
Regrading the styling, you can either wait till my changes are merged and use the styles that I wrote in `Form.css` or write your own stylesheet similar to how I made `Form.css`

**Comment by Ahmedsaed** [comment]
I suggest extracting the code into its own React component.

Extracting the code into smaller and separate components makes it easier to manage and develop. Also if my PR got merged it's going to conflict with you changes because I moved the code in `App.js` into `Resume.js` for better structure and to support other pages.

In `./src` make a new file named `PersonalInfoSection.js` (feel free to choose another name) and in the file write the following

```js
import React, { useState } from "react"; // Import useState because you are going to need it

export default function PersonalInfoSection({}) {
     // put js code here to handle state and submission
    const [userInfo, setUserInfo] = useState({
    name: "",
   email: ""
}); 

const handleChange = (e) => {
    setUserInfo({ ...userInfo, [e.target.name]: e.target.value });
  };
    
      // put html (jsx) code here. Make sure that there is only one top level element (parent)
     // Just move the form code here
    return (
      <from className="resumeSection">
        <h2>Personal Information</h2>
        <label for="name">Enter your full name: </label>
        <input type="text" class="name"></input>
        <br></br>
        <label for="email">Enter your phone number: </label>
        <input type="email" class="email"></input>
        <br></br>
        <label for="email">Enter your email address: </label>
        <input type="tel" class="phoneNumber"></input>
        <button type="submit">Save</button>
      </from>
    )
}

Update the snippet that I provided with the suggestions I wrote

**Comment by Ahmedsaed** [comment]
Now, You have a merge conflict due to a recently merged PR (Probably, my PR 😂) that conflicts with your changes. If you need assistance resolving the conflicts, we can hop on a call and work through them together.

**Comment by Epic-Eric** [comment]
## Ready to be merged
Fixes #1
Thanks to @Ahmedsaed for the help. I have implemented his advices and it is ready to be merged.

File: src/components/personal-info.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  const [userInfo, setUserInfo] = useState({
+    name: "",
+    number: "",
+    email: ""
+  })
+
+  const handleChange = (e) => {
+    setUserInfo({...userInfo, [e.target.name]: e.target.value})
+  }
+
```
Comment (Ahmedsaed) [inline src/components/personal-info.jsx:13]:
```suggestion

const handleSubmit = async (e) => {
    e.preventDefault();

    try {
      const response = await axios.post("/resume/personal", //update the route after it's implemented in the backend
      ...userInfo);
      console.log(`user info added with ID: ${response.data.id}`);
      navigateTo("mainMenu");
    } catch (error) {
      console.error("Error adding user info:", error);
      alert("Failed to add user info.");
    }
}; 
```

Add a callback function to handle submission event

File: src/components/personal-info.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  return (
+
+    <form className="resumeSection">
```
Comment (Ahmedsaed) [inline src/components/personal-info.jsx:16]:
```suggestion
    <form onSubmit={handleSubmit} className="resumeSection">
```

Finally, we assign the callback to the onSubmit event listener

**Review by Ahmedsaed — CHANGES_REQUESTED** [review]
You did a great job. I have just added some suggestions to clean up the code and handle form submission.

> [!NOTE]
> You can apply these suggestions directly by approving them through GitHub.

Just make sure that there are no typos and I also noticed that you have linter errors 

Run this command to check for linter errors
```bash
yarn prettier --check .
```

File: src/components/personal-info.jsx [hunk 4]
```diff
@@ -11,6 +11,19 @@ function PersonalInfo() {
     setUserInfo({...userInfo, [e.target.name]: e.target.value})
   }
 
+const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    try {
+      const response = await axios.post("/resume/personal", //update the route after it's implemented in the backend
```
Comment (Epic-Eric) [inline src/components/personal-info.jsx:18]: what does axios.post here?

**Comment by Epic-Eric** [comment]
Perfect! @Ahmedsaed 
Thank you so much for the corrections.

File: src/components/personal-info.jsx (hunk 4 above)
Comment (Ahmedsaed) [reply src/components/personal-info.jsx:18]: > Axios is a [promise-based](https://javascript.info/promise-basics) HTTP Client for [node.js](https://nodejs.org/) and the browser. It is [isomorphic](https://www.lullabot.com/articles/what-is-an-isomorphic-application) (= it can run in the browser and nodejs with the same codebase). On the server-side it uses the native node.js http module, while on the client (browser) it uses XMLHttpRequests.

So in other words, it's just a fancy way of sending `POST` requests (to the backend). We could have used `fetch` too.

After we are done with the development with the backend, we will run
```bash
yarn build
```

which will provide the html, css and js files which will be served by the backend.
